src/main/java/it/gov/pagopa/pu/workflow/wf/ingestionflow/debtposition/wfingestion/DebtPositionIngestionFlowWFImpl.java

```diff
@@ -122,4 +132,21 @@ private void mergeErrorDescriptions(IngestionFlowFileResult ingestionResult, Str
     }
   }
 
+  private void retrieveNoticesGenerationStatus(InstallmentIngestionFlowFileResult ingestionResult, String pdfGeneratedId) {
+    int attemptCounter = 0;
+    while (attemptCounter < MAX_ATTEMPTS &&
+      massiveNoticeGenerationStatusRetrieverActivity.retrieveNoticesGenerationStatus(ingestionResult.getOrganizationId(), pdfGeneratedId) == null) {
+      attemptCounter++;
+
+      if (attemptCounter >= MAX_ATTEMPTS) {
```

Comment:
something is strange here: you are using the same threshold to stop iterating and to `continueAsNew` the WF

The if at row 141 would never be satisfied this way!
I would split the threshold in 2 constants:
ATTEMPTS_BEFORE_WF_HISTORY_CLEAN, use at row 141
while MAX_ATTEMPTS could have this name...

let's change values instead...
MAX_ATTEMPTS=20 and 30 seconds to wait will result into a maximum wait of 10 minutes

moreover, the threshold in Temporal which would give some problem is in the order of 50k (it.gov.pagopa.pu.workflow.utilities.Constants#THRESHOLD_TEMPORAL_EVENTS_BEFORE_CONTINUE_AS_NEW)

thus, if we want to attempt so many few times, we could be not interested on reset WF history...

Reply:
Let's keep just MAX_ATTEMPTS, used to exit from the while condition.
Let's set the error_description to inform about the unavailability of the receipts

The continueAsNew logic would be not necessary: let's add a comment  before the while loop describing this:
`Not resetting Wf history through continueAsNew because MAX_ATTEMPTS is very low`

Thus, I would increase MAX_ATTEMPTS, in order to wait at least 15 minutes instead